contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsCopyStats.java

```diff
@@ -161,6 +169,41 @@ public void recordBytesWritten(long bytesWritten) {
     public void recordBytesTransferred(long bytesTransferred) {
         this.totalBytesTransferred += bytesTransferred;
         this.bytesTransferredSinceLastReport += bytesTransferred;
+        // Update the aggregated stats for total bytes transferred
+        aggStats.recordBytesTransferred(bytesTransferred);
+    }
+
+    public static void storeFetch() {
+        aggStats.storeFetch();
+    }
+    public void singleFileFetchStart(boolean isRetry) {
+        aggStats.singleFileFetchStart(isRetry);
+    }
+
+    public void singleFileFetchEnd() {
+        aggStats.singleFileFetchEnd();
+    }
+
+    public void checkSumFailed() {
+        aggStats.checkSumFailed();
+    }
+
+    public static void reportExceptionForStats(Exception e) {
+        if (ExceptionUtils.recursiveClassEquals(e, AuthenticationException.class)) {
+            aggStats.authenticateFailed();
+        } else if (e instanceof QuotaExceededException) {
+            aggStats.quotaCheckFailed();
+        } else if (e instanceof UnauthorizedStoreException) {
+            aggStats.unauthorizedStorePush();
+        } else if (e instanceof FileNotFoundException) {
+            aggStats.fileNotFound();
+        } else {
+            aggStats.fileReadFailed();
```

Comment:
This should probably tracked as an "unknown" file fetch failure, probably best stored in unknownFileFetchFailures (or similarly named).